main.go

```diff
@@ -0,0 +1,304 @@
+	discourseLog   = "      Migrated to Discourse: %s"
+	commentDone    = 2
+	commentLog     = "      Commented on issue"
+	closeDone      = 3
+	closeLog       = "      Closed GitHub issue"
+	lockDone       = 4
+	lockLog        = "      Locked GitHub issue"
+	maxCount       = 1
+	activeTpl      = "Hi %s! We are migrating our GitHub issues to Discourse (https://discuss.bitrise.io/c/issues/build-issues). From now on, you can track this issue at: %s"
+	staleTpl       = "Hi %s! We are migrating our GitHub issues to Discourse (https://discuss.bitrise.io/c/issues/build-issues). Because this issue has been inactive for more than three months, we will be closing it. If you feel it is still relevant, please open a ticket on Discourse!"
+)
+
+type RestoredIssue struct {
+	URL    string
+	Owner  string
+	Repo   string
+	IssNum int
+	Done   int
+	Extra  string
+}
+
+type Repo struct {
+	Owner string
+	Name  string
+}
+
+type Step struct {
+	LatestVersionNumber string `json:"latest_version_number"`
+	Versions            map[string]map[string]interface{}
+}
+
+type Spec struct {
+	Steps map[string]Step
+}
+
+func prefixWithRunID(str string) string {
+	return fmt.Sprintf("[TEST][%s] %s", time.Now().Format(time.RFC3339), str)
+}
+
+var c, staleCount, activeCount, cPR = 0, 0, 0, 0
+var mode string
+var f, ferr *os.File
+var accessToken string
+var ctx context.Context
+var ts oauth2.TokenSource
+var tc *http.Client
+var client *github.Client
+
+func main() {
+	mode = os.Args[1:][0]
+	var baseRepos []Repo
+
+	accessToken = os.Getenv("GITHUB_ACCESS_TOKEN")
+	ctx = context.Background()
+	ts = oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: accessToken},
+	)
+	tc = oauth2.NewClient(ctx, ts)
+	client = github.NewClient(tc)
+
+	switch mode {
+	case "test":
+		repos, _, _ := client.Repositories.List(ctx, "", &github.RepositoryListOptions{
+			Affiliation: "owner",
+		})
+		for _, r := range repos {
+			repo := Repo{r.GetOwner().GetLogin(), r.GetName()}
+			baseRepos = append(baseRepos, repo)
+		}
+	case "migrate":
+		// get spec file
+		resp, err := http.Get("https://bitrise-steplib-collection.s3.amazonaws.com/spec.json")
+		if err != nil {
+			panic(err)
+		}
+
+		// read spec file
+		spec, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		// unmarshal spec file
+		var data Spec
+		err = json.Unmarshal(spec, &data)
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		// process steps
+		for _, stp := range data.Steps {
+
+			// get latest version for step
+			url, ok := stp.Versions[stp.LatestVersionNumber]["source_code_url"].(string)
+			if !ok {
+				fmt.Println("could not convert json data")
+				os.Exit(1)
+			}
+
+			orgs := []string{
+				"bitrise-steplib",
+				"bitrise-io",
+				"bitrise-core",
+				"bitrise-community",
+				"bitrise-tools",
+				"bitrise-docker",
+				"bitrise-samples",
+			}
+
+			// filter to our repositories
+			for _, o := range orgs {
+				if strings.Contains(url, o) {
```

Comment:
please break the url into parts and check the relevant parts parts (URL_SCHEME + GIT_ORG / GIT_REPO_NAME), otherwise we migrate a step with name: `https://github.com/contributor/contibuted-step-bitrise-io`